server/channels/store/localcachelayer/post_layer.go

```diff
@@ -70,15 +70,15 @@ func (s LocalCachePostStore) InvalidateLastPostTimeCache(channelId string) {
 	}
 }
 
-func (s LocalCachePostStore) GetEtag(channelId string, allowFromCache, collapsedThreads bool) string {
+func (s LocalCachePostStore) GetEtag(channelId, userId string, allowFromCache, collapsedThreads bool) string {
 	if allowFromCache {
 		var lastTime int64
 		if err := s.rootStore.doStandardReadCache(s.rootStore.lastPostTimeCache, channelId, &lastTime); err == nil {
```

Comment:
Given now we are returning the threadMembership lastUpdated as part of the etag, the cache key can no longer be channelID right? It needs to be channelID+userID from what I understand. Otherwise, what's to stop us from returning another user's last updated for a given channelID?